src/algorithm/centroid.rs

```diff
@@ -63,25 +64,53 @@ impl Centroid for Polygon {
         if vect.len() == 1 {
             Some(Point::new(vect[0].lng(), vect[0].lat()))
         } else {
-            let mut area = 0.;
+            let area = &self.area();
             let mut sum_x = 0.;
             let mut sum_y = 0.;
             for (p1, p2) in vect.iter().zip(vect[1..].iter()) {
                 let tmp = p1.lng() * p2.lat() - p2.lng() * p1.lat();
-                area += tmp;
                 sum_x += (p2.lng() + p1.lng()) * tmp;
                 sum_y += (p2.lat() + p1.lat()) * tmp;
             }
-            area /= 2.;
             Some(Point::new(sum_x / (6. * area), sum_y / (6. * area)))
         }
     }
 }
 
+impl Centroid for MultiPolygon {
+    // See: https://fotino.me/calculating-centroids/
+    fn centroid(&self) -> Option<Point> {
+        let mut sum_x = 0.;
+        let mut sum_y = 0.;
+        let mut total_area = 0.;
+        let vect = &self.0;
+        if vect.is_empty() {
+            None
```

Comment:
You could `return None` here if you wanted to make the rest of this less indented